src/RequestHandler/outofprocess/forwardinghandler.cpp

```diff
@@ -536,103 +555,131 @@ REQUEST_NOTIFICATION_STATUS
 
 Failure:
 
-    m_fHasError = TRUE;
-    m_RequestStatus = FORWARDER_DONE;
-
-    //disbale client disconnect callback
-    RemoveRequest();
-
     //
-    // Do the right thing based on where the error originated from.
+    // Reset status for consistency.
     //
-    IHttpResponse *pResponse = m_pW3Context->GetResponse();
-    pResponse->DisableKernelCache();
-    pResponse->GetRawHttpResponse()->EntityChunkCount = 0;
-
-    if (fClientError)
+    m_RequestStatus = FORWARDER_DONE;
+    if (!m_fHasError)
     {
-        if (!m_fResponseHeadersReceivedAndSet)
+        m_fHasError = TRUE;
+
+        //
+        // Do the right thing based on where the error originated from.
+        //
+        IHttpResponse *pResponse = m_pW3Context->GetResponse();
+        pResponse->DisableKernelCache();
+        pResponse->GetRawHttpResponse()->EntityChunkCount = 0;
+
+        if (fClientError || m_fClientDisconnected)
         {
-            pResponse->SetStatus(400, "Bad Request", 0, HRESULT_FROM_WIN32(WSAECONNRESET));
+            if (!m_fResponseHeadersReceivedAndSet)
+            {
+                pResponse->SetStatus(400, "Bad Request", 0, HRESULT_FROM_WIN32(WSAECONNRESET));
+            }
+            else
+            {
+                //
```

Comment:
Get rid of the else and just make this a comment.

Reply:
yes